invoice/src/builder.rs

```diff
@@ -0,0 +1,169 @@
+// RGB wallet library for smart contracts on Bitcoin & Lightning network
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Written in 2019-2023 by
+//     Dr Maxim Orlovsky <orlovsky@lnp-bp.org>
+//
+// Copyright (C) 2019-2023 LNP/BP Standards Association. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::str::FromStr;
+
+use invoice::Network;
+use rgb::ContractId;
+
+use super::{Beneficiary, InvoiceState, Precision, RgbInvoice, RgbTransport, TransportParseError};
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct RgbInvoiceBuilder(RgbInvoice);
+
+#[allow(clippy::result_large_err)]
+impl RgbInvoiceBuilder {
+    pub fn new(beneficiary: impl Into<Beneficiary>) -> Self {
+        Self(RgbInvoice {
```

Comment:
With `Default` you will get an invalid invoice: no transports, no beneficiary; and this will require converting `finish` to return `Result` type (and introducing another error type). I am not sure this is better.